docs/adr/000x-modelling-history.md

```diff
@@ -0,0 +1,314 @@
+# X. Modelling history
+
+Date: 2018-12-17  
+Amended: 2019-01-29
+
+## Context
+
+Using terminology as defined in [ADR-3](0003-initial-domain-modelling.md):
+
+- **Document**: A piece of content in a particular locale that can have many
+versions;
+- **Revision**: A particular edit of a piece of content, represents a document at
+a specific point in time;
+- **Edition**: A particular revision that was published to GOV.UK or is the
+current draft of a document.
+
+Content Publisher currently has a concept of a document and not editions or
+revisions. A consequence of this is that an edit to a document overwrites the
+previous state of the document. Since a document can exist in two forms, draft
+and live, we can only know about one of these at any time and not of any
+previous versions.
+
+The suggestion is that we enhance Content Publisher to support the concepts of
+Document, Edition and Revision.
+
+For a document with a live version and a draft this could enable:
+
+- The ability to discard the draft and revert the current version in Content
+  Publisher back to the live version (including all images and attachments) to
+  be in sync with Publishing API;
+- The ability to have the details of the live version, most noticeably state
+  and URL, so this can be presented to users;
+- Ability to re-present either the live or draft version to Publishing API
+  should there be any problems with publishing.
+
+And with regards to history this could enable:
+
+- A record of who did what to a document and when - effectively an audit trail
+  of accountability;
+- Ability to compare two revisions of a document to show changes;
+- Storing a full history of a document - to enable importing Whitehall
+  documents with histories pre-dating the Publishing API.
+
+A [MoSCoW](https://en.wikipedia.org/wiki/MoSCoW_method) breakdown of this
+functionality is defined as:
+
+### Must
+
+- Store full information on current and live versions of a document;
+  *because* we may need to re-present either to the Publishing API as Content
+  Publisher is the source of truth.
+- Consider how a revision of a document and aspects of it have state that may
+  change;
+  *because* the concept of revision could become convoluted if it stores
+  information that a user does not directly impact.
```

Comment:
Could you give an example of such information?

Reply:
+1 to the above comment as I don't really understand this point?

Reply:
Ok got it.  Perhaps we could say something like "the concept of revision could become convoluted if it stores information that is not the content itself, such as internal state data"?

Reply:
Yup that sounds good - I've now dropped this section as it generated a lot of comments/confusion and hopefully is superfluous to the overall document.